jobs/microk8s/clusterapi/release.sh

```diff
@@ -47,24 +53,69 @@ git clone https://github.com/canonical/cluster-api-control-plane-provider-microk
 #################################################################
 if [ "${RUN_TESTS}" = true ]
 then
-    echo "Run tests"
-    sudo snap install go --classic --channel=1.18/stable
+    docker login -u ${DOCKERHUB_USR} -p ${DOCKERHUB_PSW}
+
+
+    echo "Setup management cluster"
+    sudo lxc profile create microk8s || true
+    cat "${_DIR}/microk8s.profile" | sudo lxc profile edit microk8s
+
+    sudo snap install kubectl --classic || sudo snap refresh kubectl || true
+
+    # attempt to cleanup from previous runs
+    (
+        sudo lxc exec capi-tests -- microk8s kubectl delete cluster --all --timeout=10s || true
+        sudo lxc rm capi-tests --force || true
+    )
+    sudo lxc launch ubuntu:22.04 -p default -p microk8s capi-tests
+    sleep 10
+    while ! sudo lxc exec capi-tests -- snap install microk8s --channel latest/edge --classic; do
+        sleep 3
+    done
+    sudo lxc exec capi-tests -- microk8s status --wait-ready
+    sudo lxc exec capi-tests -- microk8s enable rbac dns
+    mkdir ~/.kube -p
+    sudo lxc exec capi-tests -- microk8s config > ~/.kube/config
+
+    # download clusterctl and install
+    echo "Install clusterctl"
+    curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.4/clusterctl-linux-amd64 -o clusterctl
+    chmod +x clusterctl
+    sudo mv ./clusterctl /usr/local/bin/clusterctl
+    clusterctl version
+
+    # initialize infrastructure provider
+    echo "Initialize AWS infrastructure provider"
+    # requires sourced credentials AWS_B64ENCODED_CREDENTIALS. to refresh credentials, do
+    # export AWS_REGION=us-west-1
+    # export AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}"
+    # export AWS_ACCESS_SECRET_KEY="${AWS_ACCESS_SECRET_KEY}"
+    # clusterawsadm bootstrap iam create-cloudformation-stack
+    # export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-as-profile)
+    clusterctl init --infrastructure aws --bootstrap - --control-plane -
+
+    echo "Build Docker images and release manifests from the checked out source code"
     (
         cd bootstrap
-        make fmt
-        make vet
-        # TODO: actually do testing
-        # make lint
-        # make test
+        docker build -t cdkbot/capi-bootstrap-provider-microk8s:${RELEASE_TAG}-dev .
+        docker push cdkbot/capi-bootstrap-provider-microk8s:${RELEASE_TAG}-dev
+        sed "s,docker.io/cdkbot/capi-bootstrap-provider-microk8s:latest,cdkbot/capi-bootstrap-provider-microk8s:${RELEASE_TAG}-dev," -i bootstrap-components.yaml
```

Comment:
In the replaced string we also need the `docker.io` or else the deployment may fail, see https://github.com/canonical/cluster-api-bootstrap-provider-microk8s/pull/31 . The fact that the tests pass here may mean that this file in not read or the version of clusterctl handles this properly.